### ISSUE TITLE
Remaining Hero Properties

### ISSUE THREAD

**Issue by EgardA** [araex/BannerlordCoop#1049]
## Description

1. Sync the values in the **Synced** list below.
2. Create an integration test for all values (1 for server changing the value and 1 for client changing the value if applicable)
3. Create a command to test changing the value in game [see similar command](https://github.com/Bannerlord-Coop-Team/BannerlordCoop/blob/development/source/GameInterface/Services/Towns/Commands/TownDebugCommand.cs#L455)

Also see additional information at the bottom for more information and templates

**Synced**

| Method/Field Name | Sync Type |
|:-----------------:|:---------:|
| HitPoints | Server Sync |
| BannerItem | Server Sync |
| Issue | Server Sync |

**Deferred**

None

**Externally Deferred**

None

**Non-Synced**
None

## Intended Design

**Message networking**
![image](https://github.com/Bannerlord-Coop-Team/BannerlordCoop/assets/15619189/48f5f821-8e8f-46cd-8d73-4c2547755aea)
**Integration Testing**
![image](https://github.com/Bannerlord-Coop-Team/BannerlordCoop/assets/15619189/0e57923e-3af7-403a-aac3-aece50d7acb1)
## Location

Create a branch based from [development](https://github.com/Bannerlord-Coop-Team/BannerlordCoop)

## Related Issues

N/A

## Requirements

N/A

## Additional information

Using the [GameInterface Service Tempale](https://github.com/Bannerlord-Coop-Team/BannerlordCoop/tree/development/source/GameInterface/Services/Template) create commands to test the synced values.

For network communication use [Server Service Template](https://github.com/Bannerlord-Coop-Team/BannerlordCoop/tree/development/source/Coop.Core/Server/Services/Template) and [Client Service Template](https://github.com/Bannerlord-Coop-Team/BannerlordCoop/tree/development/source/Coop.Core/Client/Services/Template)

It is recommended to start with an integration test, you can create one using the [test template](https://github.com/Bannerlord-Coop-Team/BannerlordCoop/tree/development/source/Coop.IntegrationTests/Template).

## Definition of Done
- [ ] Class level comments exist for all new classes.
- [ ] XUnit tests exist for every method that does not require the game to be ran.
- [ ] Commands exist for all new sync items.

**Comment by AndreasOhmer** [comment]
Picking this up. Wanted to flag what I found before sending a PR, because each of the three properties has its own quirk.

## `HitPoints` — implementable, will send a PR for this one

Setter in `Hero.HitPoints` (TaleWorlds.CampaignSystem) does clamping and fires events:

```cs
set {
  ...
  // clamp to [1, MaxHitPoints]
  if (health <= WoundedHealthLimit != _health <= WoundedHealthLimit) {
    PartyBelongedTo?.MemberRoster.OnHeroHealthStatusChanged(this);
    PartyBelongedToAsPrisoner?.PrisonRoster.OnHeroHealthStatusChanged(this);
  }
  if (health > WoundedHealthLimit && IsWounded)
    CampaignEventDispatcher.Instance.OnHeroWounded(this);
}
```

That's the same pattern as other already-synced setters (HeroState etc.) — the rebroadcast on the client re-runs the side-effects locally, which is what we want for roster status. Worth confirming during in-game testing that `OnHeroWounded` doesn't double-fire anything that shouldn't (e.g. quest rewards). High-frequency updates in battle are a potential concern but the same applies to any sync; can be revisited later if it shows up as a perf issue.

## `BannerItem` — likely a no-op given existing `_battleEquipment` sync

`Hero.BannerItem` is not a backing-field property; it's a pass-through to a slot of `BattleEquipment`:

```cs
public EquipmentElement BannerItem {
  get => BattleEquipment[EquipmentIndex.ExtraWeaponSlot];
  set => BattleEquipment[EquipmentIndex.ExtraWeaponSlot] = value;
}
```

`Hero._battleEquipment` is already in the `HeroSync` property list, and `EquipmentCollectionPatches` patches `Equipment._itemSlots` array updates and publishes `ItemSlotsArrayUpdated`. So any write to `BannerItem` already propagates through that chain. Adding `BannerItem` to `HeroSync.AddProperty(...)` would duplicate the traffic without adding behavior. Suggest moving `BannerItem` to the **Deferred** (covered by `_battleEquipment` / `ItemSlotsArrayUpdated`) list in this issue, unless I'm missing a use case where it's set directly bypassing the equipment array.

## `Issue` — blocked by `IssueBase` sync

`Hero.Issue` is `public IssueBase Issue { get; private set; }`. The private setter is patchable, but the real blocker is that `IssueBase` is an abstract class with many subtypes (`BettingFraudIssue`, etc.) created via `IssueManager` / per-issue `CampaignBehavior`. For a client to honor `hero.Issue = serverIssue`, the corresponding `IssueBase` instance has to exist on the client first — i.e. issue lifecycle/registration has to be synced. I can't see any `IssueBase` registry/lifecycle handler under `GameInterface/Services` today, so this property is effectively blocked until that subsystem exists. Suggest splitting `Issue` off into its own issue and marking #1049 as blocked-by that, rather than trying to bundle it here.

## My plan

I'll send a small PR enabling **only `HitPoints`** (uncomment the existing line in `HeroSync`). If you'd rather wait until someone with an in-game test setup can verify, just say so — I'm on Linux so I can compile-check and run the existing unit/E2E suites, but I can't do multi-client in-game verification.

(Context: new contributor, three PRs in already — #1176, #1177, #1178, plus the comment on #1106.)

**Comment by AndreasOhmer** [comment]
Update / correction to [my earlier comment](https://github.com/Bannerlord-Coop-Team/BannerlordCoop/issues/1049#issuecomment-4466824393):

@NatteTosti69 pointed out that #1175 already addresses all three properties listed here. Looking at that PR:

- `HitPoints` — added to `HeroSync.AddProperty`, plus a debug command in `HeroDebugCommand`.
- `BannerItem` — handled by introducing `EquipmentElementSurrogate` (Surrogate-based serialization) rather than a separate property sync. This is cleaner than what I suggested ("already covered by `_battleEquipment`") because the surrogate handles the EquipmentElement struct directly.
- `Issue` — handled by introducing `IssueBaseSurrogate`. My earlier point that `Issue` was blocked by a missing `IssueBase` lifecycle subsystem was wrong: the Surrogate pattern sidesteps that requirement entirely by serializing the issue reference via its container rather than requiring an independent registry.

So all three properties in this issue are being addressed in #1175. I've closed my own #1179 in favor of that PR.

Sorry for the noise on this thread — the correct mental model for Bannerlord types that are awkward to lifecycle-sync (structs, abstract bases with many subtypes) is to reach for the Surrogate framework, which I didn't consider when triaging.

**Comment by garrettluskey** [comment]
Issue outside of V1 scope
